packages/okta-auth-js/lib/browser/browser.js

```diff
@@ -219,105 +218,104 @@ proto.signIn = function (opts) {
-        // Most likely cause is session does not exist or has already been closed
-        // Could also be a network error. Nothing we can do here.
-        return;
-      }
-      throw e;
-    });
+  // Clear all local tokens
+  sdk.tokenManager.clear();
+
+  if (revokeAccessToken && accessToken) {
+    await sdk.revokeAccessToken(accessToken);
   }
 
-  return allSettled([getAccessToken(), getIdToken()])
-    .then(function(tokens) {
-      accessToken = tokens[0].value;
-      idToken = tokens[1].value;
-
-      // Clear all local tokens
-      sdk.tokenManager.clear();
-
-      if (revokeAccessToken && accessToken) {
-        return sdk.token.revoke(accessToken)
-        .catch(function(e) {
-          if (e.name === 'AuthApiError') {
-            // Capture and ignore network errors
-            return;
-          }
-          throw e;
-        });
-      }
+  // No idToken? This can happen if the storage was cleared.
+  // Fallback to XHR signOut, then redirect to the post logout uri
+  if (!idToken) {
+    return sdk.closeSession()
+    .catch(function(err) {
+      // eslint-disable-next-line no-console
+      console.log('Unhandled exception while closing session', err);
     })
     .then(function() {
-      // XHR signOut method
-      if (!postLogoutRedirectUri) {
-        return closeSession();
+      if (postLogoutRedirectUri === defaultUri) {
+        window.location.reload();
+      } else {
+        window.location.assign(postLogoutRedirectUri);
       }
+    });
+  }
 
-      // No idToken? This can happen if the storage was cleared.
-      // Fallback to XHR signOut, then redirect to the post logout uri
-      if (!idToken) {
-        return closeSession()
-        .catch(function(err) {
-          // eslint-disable-next-line no-console
-          console.log('Unhandled exception while closing session', err);
-        })
-        .then(function() {
-          window.location.assign(postLogoutRedirectUri);
-        });
-      }
+  // logout redirect using the idToken.
+  var state = options.state;
+  var idTokenHint = idToken.idToken; // a string
+  var logoutUri = logoutUrl + '?id_token_hint=' + encodeURIComponent(idTokenHint) +
+    '&post_logout_redirect_uri=' + encodeURIComponent(postLogoutRedirectUri);
 
-      // logout redirect using the idToken.
-      var state = options.state;
-      var idTokenHint = idToken.idToken; // a string
-      var logoutUri = logoutUrl + '?id_token_hint=' + encodeURIComponent(idTokenHint) +
-        '&post_logout_redirect_uri=' + encodeURIComponent(postLogoutRedirectUri);
-    
-      // State allows option parameters to be passed to logout redirect uri
-      if (state) {
-        logoutUri += '&state=' + encodeURIComponent(state);
-      }
-      
-      window.location.assign(logoutUri);
-    });
+  // State allows option parameters to be passed to logout redirect uri
+  if (state) {
+    logoutUri += '&state=' + encodeURIComponent(state);
+  }
+  
+  window.location.assign(logoutUri);
```

Comment:
Browser support of assign() ?

Reply:
full support. shown here: https://www.w3schools.com/jsref/met_loc_assign.asp